### ISSUE TITLE
destrieux atlas returns wrong  region_names_

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#4330]
For the destrieux atlas, some regions are dropped (values 42 and 117 are dropped: see in `masker.labels_`). However when checking the `masker.region_names_` you see that both regions are reported to be part of the region signals that were extracted.

However 3 regions that should be in `masker.region_names_` are not:

- (148, 'R S_temporal_inf')
- (149, 'R S_temporal_sup')
- (150, 'R S_temporal_transverse')

_Originally posted by @Remi-Gau in https://github.com/nilearn/nilearn/issues/4289#issuecomment-2009408840_

**Comment by Remi-Gau** [comment]
- Should be fixed by #4329

**Comment by Remi-Gau** [comment]
Am having another issue with this:

I would expect the `labels_` or `region_names_` to be available after running fit() but they are not because they are created after transform.

This departure from our standard 'API' (underscore in attribute name means it is a fitted attribute) is a not ideal.

```python
from nilearn.datasets import fetch_atlas_destrieux_2009
from nilearn.maskers import NiftiLabelsMasker
from rich import inspect

atlas = fetch_atlas_destrieux_2009()

labels = atlas.labels.name.to_list()

for index, name in [(148, 'R S_temporal_inf'), (149, 'R S_temporal_sup'), (150, 'R S_temporal_transverse')]:
    print(f"{name in labels=}")

masker = NiftiLabelsMasker(labels_img=atlas.maps, labels=labels).fit()
print(masker.labels_)
```

```
Traceback (most recent call last):
  File "/home/remi/github/nilearn/nilearn/tmp/check_plot.py", line 14, in <module>
    print(masker.labels_)
          ^^^^^^^^^^^^^^
AttributeError: 'NiftiLabelsMasker' object has no attribute 'labels_'. Did you mean: 'labels'?
```

@mtorabi59 any thought on this?

**Comment by mtorabi59** [comment]
> Am having another issue with this:
> 
> I would expect the `labels_` or `region_names_` to be available after running fit() but they are not because they are created after transform.
> 
> This departure from our standard 'API' (underscore in attribute name means it is a fitted attribute) is a not ideal.
> 
> from nilearn.datasets import fetch_atlas_destrieux_2009
> from nilearn.maskers import NiftiLabelsMasker
> from rich import inspect
> 
> atlas = fetch_atlas_destrieux_2009()
> 
> labels = atlas.labels.name.to_list()
> 
> for index, name in [(148, 'R S_temporal_inf'), (149, 'R S_temporal_sup'), (150, 'R S_temporal_transverse')]:
>     print(f"{name in labels=}")
> 
> masker = NiftiLabelsMasker(labels_img=atlas.maps, labels=labels).fit()
> print(masker.labels_)
> ```
> Traceback (most recent call last):
>   File "/home/remi/github/nilearn/nilearn/tmp/check_plot.py", line 14, in <module>
>     print(masker.labels_)
>           ^^^^^^^^^^^^^^
> AttributeError: 'NiftiLabelsMasker' object has no attribute 'labels_'. Did you mean: 'labels'?
> ```
> 
> [@mtorabi59](https://github.com/mtorabi59) any thought on this?

So previously `labels_` was defined in `transform` and then I changed the naming to `region_names_`. I believe the reason for this is first of all, `labels_` or `region_names_` is usually not useful before transform and `region_names_` is returned besides `region_signals` so that the user knows which signal corresponds to which region. And second, before `transform` you can simply use `labels` I guess. Also, `region_ids` and `region_atlas_` are also defined after `transform` and they cannot be defined before that because we can get them only after `_filter_and_extract` is called and for `region_names_` we need `region_ids`. The reason we need `_filter_and_extract` which actually runs `img_to_signals_labels` is because of masking. Does that make sense?
